docs/design/2020-06-24-placement-rules-in-sql.md

```diff
@@ -735,28 +563,86 @@ As database IDs are all globally unique, it's fine to replace table ID with data
+3. When a placement rule is added on partition `p0` of `t`, all data on `p0` is placed based on the rule of `p0`, but other partitions stay still.
+4. When the placement rule on `p0` is removed, data on `p0` is placed based on the rule of `t`, just like other partitions.
+
+Rules priorities are checked when a placement rule is added, altered, or dropped.
+
+Rule priorities can be implemented by fields `index` and `override` in the PD placement rule configuration. `override` is alway enabled, and `index` stands for the priority. Rules with higher `index` will overwrite the rules with lower `index` and same key range, but rules with same `index` don't overwrite each other, they just accumulate.
+
+Specifically, `index` is in such a format:
+
+* `index` of default placement rules is 0
+* `index` of database placement rules is 1
+* `index` of table placement rules is 2
+* `index` of partition placement rules is 3
+* `index` of index placement rules is 4
+
+In such a way, the most granular rule always works.
+
+## Impact & Risks
+
+1. The largest risk is designing a set of SQL statements that are sufficiently flexible to handle major use cases, but not too flexible that misconfiguration is likely when the user has compliance requirements. The following design decisions have been made to mitigate this risk:
+  - The DDL statement to `ALTER TABLE t1 ADD PLACEMENT` has been removed from the proposal.
+  - The DDL statement `CREATE PLACEMENT POLICY` has been added (allowing common configurations to be saved).
+  - Configuring placement rules for indexes is no longer supported (we can add it once global indexes are added).
+  - The inheritance rules have been simplified to match character set/collation inheritance.
+2. There is a risk that we do not fully understand compliance requirements (see unresolved questions), or that the substaintial effort required to achieve compliance with DDL, internal SQL and DML.
+3. Related to the above point, there is risk that the implementation of compliance requirements has significant burden on multiple teams, including ecosystem tools (Backup, CDC, DM). Even tools such as dumpling should ideally be able to backup placement rules in logical form.
+4. There is some risk that a common use case can not be expressed in `CONSTRAINT` syntax, leading to complicated scenarios where users still need to express placement by using PD directly. Ideally, we can recommend users use SQL rules exclusively.
+5. Many other features in TiDB are in development, some of which may influence placement rules. Clustered index affects the key format of primary index. Fortunately, the prefix of key range is untouched. Global secondary index largely affect the placement rules of partitioned tables. The key range of one global secondary index is not successive, so if it's necessary to define placement rules on the index, multiple rules should be generated in the PD. But for now, there's no such scenario.
```

Comment:
Is not the range of one global secondary index successive?
From above:
```
* Unpartitioned index: the concatenation of table id and index id, e.g. `100_1`
```
Or how do you mean?